enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -0,0 +1,77 @@
+import type { Query } from "history";
+import { useState } from "react";
+
+import type { SdkClickActionPluginsConfig } from "embedding-sdk";
+import { InteractiveAdHocQuestion } from "embedding-sdk/components/public/InteractiveDashboard/InteractiveAdHocQuestion";
+import { getNewCardUrl } from "metabase/dashboard/actions/getNewCardUrl";
+import type { NavigateToNewCardFromDashboardOpts } from "metabase/dashboard/components/DashCard/types";
+import { useStore } from "metabase/lib/redux";
+import { getMetadata } from "metabase/selectors/metadata";
+import type { DashboardId, QuestionDashboardCard } from "metabase-types/api";
+
+import { StaticDashboard } from "../StaticDashboard";
+
+export type InteractiveDashboardProps = {
+  dashboardId: DashboardId;
+  initialParameterValues?: Query;
+  withTitle?: boolean;
+  withDownloads?: boolean;
+  hiddenParameters?: string[];
+  questionHeight?: number;
+  questionPlugins?: SdkClickActionPluginsConfig;
+};
+
+export const InteractiveDashboard = (props: InteractiveDashboardProps) => {
+  const { dashboardId, withTitle, questionHeight, questionPlugins } = props;
+
+  const store = useStore();
+
+  const [adhocQuestionUrl, setAdhocQuestionUrl] = useState<string | null>(null);
+
+  const handleNavigateToNewCardFromDashboard = ({
+    nextCard,
+    previousCard,
+    dashcard,
+    objectId,
+  }: NavigateToNewCardFromDashboardOpts) => {
+    const state = store.getState();
```

Comment:
Curious about why we are using `store.getState` as opposed to using selectors here? Is there a performance-related reason?

Reply:
fair enough, I just realized that you are also using the `getMetadata` from selectors below, so this is good.